### PR TITLE
Issue 9547 - Update IPA to IPA migration design doc

### PR DESCRIPTION
**Pull request by mreynolds389**
Update the ipa to ipa migration doc in regards to the new IPA migration tool

Fixes: https://pagure.io/freeipa/issue/9547

### CODE REVIEW

**Comment by flo-renaud** [comment]
@mreynolds389 the failure of `test_commands` is related to feb 29, please see [9548](https://pagure.io/freeipa/issue/9548). Re-run the test tomorrow and it should succeed.

**Review by abbra — CHANGES_REQUESTED** [review]
I have one small inline comment and larger, here. 

I think it will be more clear if instead of comma-prefixed DNs you'd write a 'subtree of ...', e.g. 

> Topology (,cn=topology,cn=ipa,cn=etc,$SUFFIX)

would become

> Topology (subtree of `cn=topology,cn=ipa,cn=etc,$SUFFIX`)

This syntax (DN with preceding `,`) is not described anywhere in the document and is not used elsewhere in IPA design documents (nor in LDAP specifications), so it is not immediately clear what does it mean.

File: doc/designs/ipa_to_ipa_migration.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # IPA Migration
 
-## Overview
+{% include toc.md %}
```
[the remark this answers is not in the document]
Comment (mreynolds389) [reply doc/designs/ipa_to_ipa_migration.md:3]: > this include is not needed. ReadTheDocs automatically adds table of contents.

Great, good to know!

**Comment by mreynolds389** [comment]
> I have one small inline comment and larger, here.
> 
> I think it will be more clear if instead of comma-prefixed DNs you'd write a 'subtree of ...', e.g.
> 
> > Topology (,cn=topology,cn=ipa,cn=etc,$SUFFIX)
> 
> would become
> 
> > Topology (subtree of `cn=topology,cn=ipa,cn=etc,$SUFFIX`)
> 
> This syntax (DN with preceding `,`) is not described anywhere in the document and is not used elsewhere in IPA design documents (nor in LDAP specifications), so it is not immediately clear what does it mean.

Right it's not a specification, but it's how it's used in the tool's code.  There are comments in the doc stating that if it starts with a comma then it's a subtree.  Regardless I will gladly change the document to make it more clear

**Comment by mreynolds389** [comment]
> > I have one small inline comment and larger, here.
> > I think it will be more clear if instead of comma-prefixed DNs you'd write a 'subtree of ...', e.g.
> > > Topology (,cn=topology,cn=ipa,cn=etc,$SUFFIX)
> > 
> > 
> > would become
> > > Topology (subtree of `cn=topology,cn=ipa,cn=etc,$SUFFIX`)
> > 
> > 
> > This syntax (DN with preceding `,`) is not described anywhere in the document and is not used elsewhere in IPA design documents (nor in LDAP specifications), so it is not immediately clear what does it mean.

@abbra changes made, please review

**Comment by mreynolds389** [comment]
@abbra I don't have authorization to merge :-)

**Comment by abbra** [comment]
@mreynolds389 don't worry, this will be handled by the people who have those rights. We do not use github for this, github repo is only for raising PRs.

**Comment by flo-renaud** [comment]
As the original PR adding the migration doc (#7078) was backported to ipa-4-10 and ipa-4-11, I am also adding the labels.

**Comment by flo-renaud** [comment]
master:

* 557f0a5639e65b952ed0ce82e4ef42683bf75178 Issue 9547 - Update IPA to IPA migration design doc

**Pull request closed without merging**